content/blog/build-and-release-hugo-blog-site-using-azure-pipelines.md

```diff
@@ -0,0 +1,208 @@
+
+I used the instructions on [Hosting Hugo site on GitHub](https://gohugo.io/hosting-and-deployment/hosting-on-github/) to create 2 git repositories.  
+1. A `blog` repository containing my markdown files, Hugo templates, theme, etc.  
+2. A `<username>.github.io` repository that is the source of my GitHub Pages site.
+
+The `ronaldbosma.github.io` repository was included as a submodule in the `blog` repository. This allowed me to run the Hugo command and generate the static website in the `ronaldbosma.github.io` repository. The image below shows a graphical representation of this.
+
+![Initial repository setup](../../static/images/build-and-release-hugo-site-using-azure-pipelines/initial-repo-setup.png)
+<!-- ![Initial repository setup](../../../../../images/build-and-release-hugo-site-using-azure-pipelines/initial-repo-setup.png) -->
+
+As per the instructions I used a `deploy.sh` script to generate the Hugo static website in the public folder and push any changes to `ronaldbosma.github.io`. This has a few disadvantages though.
+
+First off it's a manual action which can be forgotten easily. Also, I'm a developer and I dislike doing things manually.
+
+I also prefer to create separate branches for my posts so I can create a pull request and ask my peers for feedback. The `deploy.sh` script can be run from every branch. Possibly publishing something I don't want to or removing changes I haven't merged from master yet.
+
+The last problem I had was that for some reason the link to the submodule seemed to disappear. Making pushing any changes to `ronaldbosma.github.io` impossible. As a pragmatic solution I just removed and added the submodule manually. Which was cumbersome.
+
+What I wanted was to automatically build and release my blog whenever I push a change in the master branch of my `blog` repository. I used Azure Pipelines to make this happen. The image below shows the various steps that are executed in the pipeline.
+
+![Pipeline](../../static/images/build-and-release-hugo-site-using-azure-pipelines/pipeline.png)
+<!-- ![Pipeline](../../../../../images/build-and-release-hugo-site-using-azure-pipelines/pipeline.png) -->
+
+First a build pipeline is triggered whenever I push to master. Using a Hugo build task the static site is generated and published as an artifact of the build. When the build succeeds a release pipeline is triggered which will take the site and push this into the `ronaldbosma.github.io` repository using a Publish to GitHub Pages task.
+
+Although I'm interacting with a git repository when I publish my site, I decided to put this step in a release pipeline and not a build pipeline. I'm releasing my site here. Not building it. This also allows me to add more stages to for example a testing environment or change the target location and type of my site altogether without affecting my build pipeline.
```

Comment:
stages **to** for example. Deze zin loopt niet zo lekker, maar ik weet ook niet zo goed hoe ik de zin anders zou formuleren...